admin/controller/extension/payment/razorpay.php

```diff
@@ -192,4 +194,1091 @@ protected function validate()
+				$url .= '&filter_entity_id=' . $this->request->get['filter_entity_id'];
+			}
+            $data['breadcrumbs'] = array();
+
+        $data['breadcrumbs'][] = array(
+        'text' => $this->language->get('text_home'),
+        'href' => $this->url->link('common/dashboard', 'user_token=' . $this->session->data['user_token'], true)
+        );
+
+        $data['breadcrumbs'][] = array(
+        'text' => $this->language->get('subscription_title'),
+        'href' => $this->url->link('extension/payment/razorpay/getSubscription', 'user_token=' . $this->session->data['user_token'] . $url, true)
+        );
+        //Invoice 
+        $api = $this->getApiIntance();
+        $data['invoiceDetails'] = array();
+        $data['invoiceResult']=$invoiceResult = $api->invoice->all(["subscription_id"=>$results['subscription_id']]);
+       
+        foreach ($invoiceResult['items'] as $result) {
+            $data['invoiceDetails'][] = array(
+                'id'      => $result['id'],
+                'recurring_amt'=> $result['line_items'][0]['net_amount']/100,
+                'addons'=>$result['line_items'][0]['unit_amount']/100,
+                'status'=>ucfirst($result['status']),
+                'total_amt'=>$result['amount']/100,
+                'date'=>date('M d, Y', $result['billing_start']),
+                'short_url'=>$result['short_url']
+            );
+        }
+      
+        $data['singleResume']= $this->url->link('extension/payment/razorpay/changeSingleStatus', 'user_token=' . $this->session->data['user_token'] . '&entity_id=' . $results['sub_id'] . '&status=1'. $url, true);
+        $data['singlePause'] = $this->url->link('extension/payment/razorpay/changeSingleStatus', 'user_token=' . $this->session->data['user_token'] . '&entity_id=' . $results['sub_id'] . '&status=2'. $url, true);
+          
+        $data['singleCancel'] = $this->url->link('extension/payment/razorpay/changeSingleStatus', 'user_token=' . $this->session->data['user_token'] . '&entity_id=' . $results['sub_id'] . '&status=3'. $url, true);
+        $data['back'] = $this->url->link('extension/payment/razorpay/getSubscription', 'user_token=' . $this->session->data['user_token'] . $url, true);
+        $data['header'] = $this->load->controller('common/header');
+		$data['column_left'] = $this->load->controller('common/column_left');
+		$data['footer'] = $this->load->controller('common/footer');
+
+        $this->response->setOutput($this->load->view('extension/payment/razorpay_subscription_info', $data));
+
+        }
+        else {
+			return new Action('error/not_found');
+		}
+    }
+
+    public function install()
+    {
+        $this->load->model('extension/payment/razorpay');
+        
+        $this->model_extension_payment_razorpay->createTables();
+    }
+
+    public function uninstall()
+    {
+        $this->load->model('extension/payment/razorpay');
+
+        $this->model_extension_payment_razorpay->dropTables();
+    }
```

Comment:
Move this set of code to beginning of the file

Reply:
NA